salt/config/__init__.py

```diff
@@ -3811,10 +3781,31 @@ def apply_minion_config(overrides=None,
 
     # Check and update TLS/SSL configuration
     _update_ssl_config(opts)
+    _update_discovery_config(opts)
 
     return opts
 
 
+def _update_discovery_config(opts, minion=True):
```

Comment:
We actually don't need to use an argument here to distinguish master from minion. `apply_master_config()` sets `opts['__role'] = 'master'`, so we can just get rid of the `minion` argument here.